### PR TITLE
[Truffle] Add name to Thread layout

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

File: truffle/src/main/java/org/jruby/truffle/core/thread/ThreadLayout.java [hunk 1]
```diff
@@ -46,7 +46,8 @@ DynamicObject createThread(
             @Nullable @Volatile Object value,
             AtomicBoolean wakeUp,
             @Volatile int priority,
-            @Nullable DynamicObject threadGroup);
+            @Nullable DynamicObject threadGroup,
```
Comment (eregon) [inline truffle/src/main/java/org/jruby/truffle/core/thread/ThreadLayout.java]: BTW, threadGroup no longer needs to be @Nullable, right?

File: truffle/src/main/ruby/core/thread.rb [hunk 2]
```diff
@@ -193,6 +193,21 @@ def priority=(val)
     Kernel.raise ThreadError, "Thread#priority= primitive failed"
   end
 
+  def name
+    Truffle.primitive :thread_get_name
+    Kernel.raise ThreadError, "Thread#priority primitive failed"
```
Comment (eregon) [inline truffle/src/main/ruby/core/thread.rb]: Typo

File: truffle/src/main/java/org/jruby/truffle/core/thread/ThreadLayout.java (hunk 1 above)
Comment (bjfish) [inline truffle/src/main/java/org/jruby/truffle/core/thread/ThreadLayout.java]: Yes, I've removed the `@Nullable`.

File: truffle/src/main/ruby/core/thread.rb (hunk 2 above)
Comment (bjfish) [inline truffle/src/main/ruby/core/thread.rb]: Typo fixed.

**Pull request merged**